### DISCUSSION TITLE
Storing data in an item?

### DISCUSSION THREAD

**Discussion by altooma** [FabricMC/fabric-api#1745] [Mod Dev Support]
I'm looking to create an item that generates a unique quest (e.g kill 10 zombies) on the items creation in the world. This would mean that each item of the same class would have a randomly generated quest associated to its individual data, using nbt or some other method? 

The item would be found in loot tables or bought from villagers so when it is generated in these events the random quest would be assigned to the item. This is just a broad concept and I am interested in any way that this could be achieved.

**Comment by warjort** [comment]
Data for items goes in the ItemStack. It lets you have custom NBT tags for storing data.
See ````ItemStack.getOrCreateSubTag(String)````

For the random quest data, I guess you will need to write your own LootFunction and register it.
See what vanilla does in ````net.minecraft.loot.function.LootFunctionTypes````, 
e.g. minecraft:enchant_randomly or minecraft:set_stew_effect

**Comment by kanpov** [comment]
I've created a [library](https://github.com/RedGrapefruit09/ItemNBT) that provides much more features than simple fetching data from an `NbtCompound`.

You might want to check it out too, @altooma .
